ProceduralForest/Structures/Renderer.cpp

```diff
@@ -0,0 +1,148 @@
+//
+// Created by Steven Smith on 2020-04-04.
+// A class created to help and simplify the main driver into something more simple
+// and easier to manage. A singleton instance that would keep references to important objects such as
+// the currently active Shader
+//
+#include "Renderer.h"
+#include "WindowManager.h"
+//default constructor
+
+std::vector<Shader*> Renderer::shaderList;
+Shader * Renderer::currentShader = nullptr;
+GLuint Renderer::renderMode = -1;
+void Renderer::Initialize(glm::vec3 color) {
+    glfwMakeContextCurrent(WindowManager::getWindow());
+
+    // Initialize GLEW
+    glewExperimental = true; // Needed for core profile
+    if (glewInit() != GLEW_OK) {
+        fprintf(stderr, "Failed to initialize GLEW\n");
+
+        getchar();
+        exit(-1);
+    }
+
+    // Somehow, glewInit triggers a glInvalidEnum... Let's ignore it
+    glGetError();
+
+    // Black background
+    glClearColor(color.x, color.y, color.z, 0.0f);
+
+    // Enable depth test
+    glEnable(GL_DEPTH_TEST);
+    glDepthFunc(GL_LESS);
+    glEnable(GL_CULL_FACE);
+
+
+    CheckForErrors();
+}
+
+void Renderer::Shutdown() {
+    // Shaders
+    for (std::vector<Shader*>::iterator it = shaderList.begin(); it < shaderList.end(); ++it)
+    {
+        glDeleteProgram((*it.base())->ID);
+    }
+    currentShader = nullptr;
+    shaderList.clear();
+    WindowManager::Shutdown();
+}
+void Renderer::BeginFrame()
+{
+    // Clear the screen
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+}
+
+void Renderer::EndFrame()
+{
+    // Swap buffers
+    glfwSwapBuffers(WindowManager::getWindow());
+}
+
+
+bool Renderer::PrintError()
+{
+    static bool checkForErrors = true;
+
+    //
+    if( !checkForErrors )
+    {
+        return false;
+    }
```

Comment:
Currently always false, though perhaps will be used later? Either way doesn't affect code.